lib/Doctrine/ORM/Mapping/DefaultEntityListenerResolver.php

```diff
@@ -46,7 +45,7 @@ public function register($object)
             throw new InvalidArgumentException(sprintf('An object was expected, but got "%s".', gettype($object)));
         }
 
-        $this->instances[get_class($object)] = $object;
+        $this->instances[$object::class] = $object;
```

Comment:
Same as mentioned before. I won't comment the other lines with `$variable::class`